accounts-db/src/tiered_storage/hot.rs

```diff
@@ -457,25 +493,148 @@ impl HotStorageWriter {
             storage: TieredStorageFile::new_writable(file_path)?,
         })
     }
+
+    /// Persists an account with the specified information and returns
+    /// the stored size of the account.
+    fn write_account(
+        &self,
+        lamports: u64,
+        rent_epoch: Epoch,
+        account_data: &[u8],
+        executable: bool,
+        account_hash: Option<&AccountHash>,
+    ) -> TieredStorageResult<usize> {
+        let optional_fields = AccountMetaOptionalFields {
+            rent_epoch: (rent_epoch != Epoch::MAX).then_some(rent_epoch),
+            account_hash: account_hash.map(|account_hash| *account_hash),
+        };
+
+        let mut flags = AccountMetaFlags::new_from(&optional_fields);
+        flags.set_executable(executable);
+
+        let padding_len = padding_bytes(account_data.len());
+        let meta = HotAccountMeta::new()
+            .with_lamports(lamports)
+            .with_account_data_size(account_data.len() as u64)
+            .with_account_data_padding(padding_len)
+            .with_flags(&flags);
+
+        let mut stored_size = 0;
+
+        stored_size += self.storage.write_pod(&meta)?;
+        stored_size += self.storage.write_bytes(account_data)?;
+        stored_size += self
+            .storage
+            .write_bytes(&PADDING_BUFFER[0..(padding_len as usize)])?;
+        stored_size += write_optional_fields(&self.storage, &optional_fields)?;
+
+        Ok(stored_size)
+    }
+
+    /// A work-in-progress function that will eventually implements
+    /// AccountsFile::appends_account()
+    pub fn write_accounts<
+        'a,
+        'b,
+        T: ReadableAccount + Sync,
+        U: StorableAccounts<'a, T>,
+        V: Borrow<AccountHash>,
+    >(
+        &self,
+        accounts: &StorableAccountsWithHashesAndWriteVersions<'a, 'b, T, U, V>,
+        skip: usize,
+    ) -> TieredStorageResult<()> {
+        let mut footer = new_hot_footer();
+        let mut index = vec![];
+        let mut cursor = 0;
+
+        // writing accounts blocks
+        let len = accounts.accounts.len();
+        for i in skip..len {
+            let (account, address, account_hash, _write_version) = accounts.get(i);
+            let index_entry = AccountIndexWriterEntry {
+                address,
+                offset: HotAccountOffset::new(cursor)?,
+            };
+            if let Some(account) = account {
```

Comment:
Looks like it is a reference so that it disallows me to call `unwrap_or_default()` without cloning.

Are there ways to workaround this?  Or simply just pass default value as you suggested above?

```
    |
559 |             let account = account.unwrap_or_default();
    |                           ^^^^^^^ the trait `std::default::Default` is not implemented for `&T`
    |
help: consider further restricting this bound
    |
539 |         T: ReadableAccount + Sync + Default + Clone,
    |                                             +++++++
help: consider using clone here
    |
559 |             let account = account.clone().unwrap_or_default();
    |                                  ++++++++
```